Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Gun.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+
+}
+
+bool Gun::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::Shoot(VDir target,id_ptr_on<Human> shooter)
+{	
+    if(Handle_Ammo())
+    {
+      if(GetOwner()->GetOwner().valid())
+      {
+          id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner()->GetOwner());
+          p->MakeMovementPattern(target,shooter);
+      }
+             //       Create<Item>(Bullet_Casing::T_ITEM_S(),GetOwner);	
+    }
+    else
+    {
+          PlaySoundIfVisible("empty.ogg",owner.ret_id());
+		// the *click* text from ss13
```

Comment:
@shelltitan I have tested locally sounds and my variant works for me.
My test code is 
```
        id_ptr_on<IOnMapObject> item = Network2::ExtractObjId(obj);
        if (item && item->GetOwner())
        {
            if (id_ptr_on<Crowbar> crowbar = interface_.GetActiveHand().Get())
            {
                crowbar->PlayTestSound();
            }
            // etc
        }
```
```
void Crowbar::PlayTestSound()
{
    PlaySoundIfVisible("door_open.ogg", GetOwner()->GetOwner().ret_id());
}
```
I was able to hear the sound. So please check everything carefully.
`GetOwner()->GetOwner().ret_id()` with `ret_id` should be used (without it will not work)